src/picterra/client.py

```diff
@@ -725,5 +766,96 @@ def train_detector(self, detector_id):
             detector_id (str): The id of the detector
         """
         resp = self.sess.post(self._api_url('detectors/%s/train/' % detector_id))
-        assert resp.status_code == 201, resp.status_code
+        if not resp.ok:
+            raise APIError(resp.text)
         self._wait_until_operation_completes(resp.json())
+
+
+    def run_advanced_tool(self, tool_id: UUID, inputs: dict, outputs: dict):
+            """
```

Comment:
nit: is the indentation correct here? It looks like it's inconsistent with the other methods (8 spaces rather than 4?)

Reply:
Fixed